Review and discuss quickpoll meraki script proposed changes

compass259 opened the pull request:
Review and discuss quickpoll meraki script proposed changes

compass259 committed:
Add files via upload

compass259 committed:
Update meraki_status_quickpoll.sh

compass259 committed:
Update meraki_vpnstats_quickpoll.sh

compass259 committed:
Update meraki.sh

Modifications of meraki.sh file to target the data generated by the meraki_status_quickpoll.sh script.

compass259 committed:
Add files via upload

This is an updated Zabbix Meraki template for my purposes.
1. Disable the existing "{HOSTNAME} data pull item".  The "meraki_status.sh" script is perfect for the initial device creation and slow polling, but less so for quick 5 min polling since the script takes a while to finish.
2. New "{HOSTNAME} data pull - uplink" bulk data from the "meraki_quickpull.json" file.  Dependent items have been created to populate data from this bulk data pull.  Items are uplink status, cellular signal strength (RSRP), cellular signal strength (RSRQ), uplink packet loss, uplink latency.  All of these items are currently being populated every five minutes in the template.  The "meraki_quickpull.sh script" was kept to API calls that would finish quickly (approx. 75 seconds), so that he dependent items could be populated reliably every five minutes to give good trending and historical data.
3. A new "{HOSTNAME} data pull - vpn statistics" bulk pull item was also added for the "meraki_vpnstats_quickpoll.sh" to pull VPN statistics.  I meant to include this in the "meraki_quickpull.sh" script, but it made the data download too long (approx. 220 secs) for the intended five min check.  I have not yet created a dependent item for the VPN statistics values, however I intend to do so as a 15 or 30 minute template poll.

compass259 commented:
This is an updated Zabbix Meraki template as a proposal.  This fits my Zabbix implementation purpose well.
1. Disable the existing "{HOSTNAME} data pull item".  The "meraki_status.sh" script is perfect for the initial device creation and slow polling, but less so for quick 5 min polling since the script takes a while to finish.
2. New "{HOSTNAME} data pull - uplink" bulk data from the "meraki_quickpull.json" file.  Dependent items have been created to populate data from this bulk data pull.  Items are uplink status, cellular signal strength (RSRP), cellular signal strength (RSRQ), uplink packet loss, uplink latency.  All of these items are currently being populated every five minutes in the template.  The "meraki_quickpull.sh script" was minimized to API calls that would finish quickly (approx. 75 seconds), so that he dependent Meraki items could be populated reliably every five minutes to give good trending and historical data.
3. A new "{HOSTNAME} data pull - vpn statistics" bulk pull item was also added for the "meraki_vpnstats_quickpoll.sh" to pull VPN statistics.  I meant to include this in the "meraki_quickpull.sh" script, but it made the API data download procedure too long (approx. 220 secs) for the intended five min check.  I have not yet created a dependent item for the VPN statistics values, however I intend to do so as a 15 or 30 minute template poll.

zdegraaf commented on meraki_vpnstats_quickpoll.sh:
```diff
@@ -0,0 +1,53 @@
+
+#NOTES:
+#This Meraki API data pull script was created specifically to generate a data file containing device VPN tunnel statistics that Zabbix can poll periodically.  
+#The complete script takes roughly 180 seconds to complete for me.
```
I would guess that the script takes longer the more devices it's pulling data for correct? Could you change this comment to reflect roughly how long this script takes to run per monitored device? That way there is context for how long things take to run without revealing anything about your environment. I have no idea how long it actually takes, but as an example of what it could look like, 

The script takes approximately 10 seconds to run for every 100 VPN endpoints.

compass259 replied:
Comment changed.  Thank you for that suggestion.

zdegraaf commented on meraki_status_quickpoll.sh:
```diff
@@ -0,0 +1,60 @@
+
+#NOTES:
+#This Meraki cloud API data pull script was created to generate an API aggregate data file that Zabbix can poll periodically.  
+#The complete script takes approximately 75 seconds to complete for me.
```
Let's change this comment to the same x seconds per y devices as the others

compass259 replied:
Comment changed also.

zdegraaf reviewed (changes requested):
So overall I am very intrigued by the additions and changes, they look like they will help the the script run quicker and be more compatible with various deployments. I put two comments in for changes in the script comments that I would like to adjust before merging this back up.

compass259 committed:
Update meraki_status_quickpoll.sh

compass259 committed:
Update meraki_vpnstats_quickpoll.sh

compass259 commented:
Thanks you for those suggestions for the timings, you are correct that could provide hints as to the size of my environment. 
 I have changed the comments as suggested.  Please note that this is still a work in progress for me.  I have adjusted things slightly since to try a few things.  I'll post more updates as I have them, but this is essentially the core of it.

zdegraaf commented:
So when I imported the template you adjusted, I am only seeing triggers for the "Network Status Check that I created, but you've disabled in your template. Was this intentional?

zdegraaf commented:
So after thinking about this, I added a dev-testing branch to the repository. Especially since the changes you are making are a work in progress I think it would be best to change the pull request to point at that branch. This should give us the opportunity to collaborate a bit on the changes until they are production ready, while also letting us make changes a bit more freely without compromising the main branch. 

I am going to close this pull request, but go ahead and resubmit it and I will merge it in straight away and we can work on any kinks together.

compass259 commented:
I found that the device status and the interface status are the only status checks that are of value to me.  I haven't seen the value of the other status checks, especially at the slow rate of the first script (15 mins plus run time).  Have you seen instances where those others have been useful?
I added the two status checks I want to keep to be dependent on the quickpoll 5mins script so that I have reasonably accurate and up to date uptime and status history info.

I was trying to determine what was actually needed for your host creation process script, but I haven't had the time to fully explore that. That is partly why some are disabled. 
I plan on running the original meraki_status.sh script every 12 hours or so for host discovery and creation only, and the quickpoll script every 5 mins for the checks that should be more often.
One of the changes I made was to create a seventh Wan connection item prototype for the interface type (wan1 or cellular), and I removed the {host} and {interface} variables from the other item prototype names.  I am exporing a way to have a screen or dashboard that would display Meraki devices with packet loss.  
The monitor-overview screen is much more readable when the column names are the same (Device status check, Interface latency, etc...).  Now I just have to figure out how to generate some type of display that will only show devices that have packet loss.  I was toying with the idea of having the 5 min check for packet loss add or remove a host group to the Meraki device called "Meraki Packet Loss" so that I could have the Overview monitoring display filtered by that host group, but I think ot might be too much messing around with the host groups.


Sent from Rogers Yahoo Mail for iPhone


On Tuesday, May 4, 2021, 12:13 PM, zdegraaf ***@***.***> wrote:




So when I imported the template you adjusted, I am only seeing triggers for the "Network Status Check that I created, but you've disabled in your template. Was this intentional?

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.
